### PR TITLE
Update headercard to use filter to add ExPlat parameter

### LINKED ISSUE #8227: Add `woocommerce_admin_install_timestamp` to ExPlat API calls
### Description

We don't have a good way to detect a new store for the Experimentation Platform since we [don't support WPCOM auth](https://github.com/woocommerce/woocommerce-admin/issues/8071). New store criteria for eligibility check is critical since it's a common use-case for experimenters.

A workaround I propose is to add the value stored in `woocommerce_admin_install_timestamp` to the ExPlat integrations:

**PHP integration:**
https://github.com/woocommerce/woocommerce-admin/blob/281ae241d0c6d15f696c74cd6bd28294dee16a0f/includes/class-experimental-abtest.php#L158-L162

**JS integration:**
https://github.com/woocommerce/woocommerce-admin/blob/281ae241d0c6d15f696c74cd6bd28294dee16a0f/packages/explat/src/assignment.ts#L21-L29

An appropriate parameter name would be `woo_wcadmin_install_timestamp`.

### PR DESCRIPTION
**Pull request by ilyasfoo**
Fixes #8227

This PR:
1. Adds the capability for tasklist progression experiments to append WCA's install timestamp to ExPlat API request
2. Changes the way experiment is called. Instead of using `useExperiment`, we're using a custom hook since we need to conditionally load the experiment after the required option is loaded

After a while of testing, I observed a noticeable delay in my local for the experiment tasklist to be displayed since it's not preloaded. We may need to consider preloading `woocommerce_admin_install_timestamp` option and hydrate if this is proven to be a problem.

### Detailed test instructions:

1. Since we're using dynamic experiments name, this needs to be tested anywhen in February 2022. Alternatively, hardcode the month [here](https://github.com/woocommerce/woocommerce-admin/blob/6824e721c21d01f9f635fe0c9359f0046fe4ba13/client/homescreen/hooks/use-headercard-experiment-hook.js#L46) to `'02'`
1. Use the treatment bookmarklet for experiment `woocommerce_tasklist_progression_headercard_2col_2022_02`
2. Open your browser's console network tab
3. Go to WooCommerce > Home
4. Observe the network request for `https://public-api.wordpress.com/wpcom/v2/experiments/0.1.0/assignments/woocommerce?experiment_name=woocommerce_tasklist_progression_headercard_2col_2022_02` contains the payload `woo_wcadmin_install_timestamp` and the value is a timestamp
5. If you received treatment variation, you should see the two column tasklist experiment

### CODE REVIEW

**Review by louwie17 — CHANGES_REQUESTED** [review]
> After a while of testing, I observed a noticeable delay in my local for the experiment tasklist to be displayed since it's not preloaded. We may need to consider preloading woocommerce_admin_install_timestamp option and hydrate if this is proven to be a problem.

I encountered the same thing, probably because I have XDebug enabled locally, so things are slower in general. But this causes some real jankiness, with showing the normal task list first:
task-list-experiment-timestamp

I think pre-loading or making it available through `wcSettings.admin` would be the best bet. I would lean towards the latter.

**Comment by ilyasfoo** [comment]
Thanks for reviewing this, @louwie17! I agree, I've used the preload option for Homescreen.php [015e0b9](https://github.com/woocommerce/woocommerce-admin/pull/8233/commits/015e0b907fbee993dc09b46573df8b0241d843c0) since it's already available.

Please have another look!

**Review by adrianduffell — COMMENTED** [review]
This tested well for me! I'll leave it up to @louwie17 to have final review.

**Review by louwie17 — CHANGES_REQUESTED** [review]
Thanks for the update @ilyasfoo, things are feeling better now, although there is still a bit of a flash between rendering the placeholder of the old task list and then the new task list.
I can't put my finger on what code exactly is causing this, but this seems to be the flow:
- Loading icon happens until the experiment is loaded
- Experiment is loaded and the inbox panel placeholder shows quickly with the Stats overview panel
- Then it loads the new experiment task list
- Clearing the UI and showing a loading spinner
- Then shows the placeholder of the new task list

task-list-experiment

After writing these steps out I am thinking we might have to add some loading logic here: https://github.com/woocommerce/woocommerce-admin/blob/main/client/homescreen/layout.js#L167-L171
Or make sure that whatever loading icon takes up the entire screen afterwards doesn't.

It be nice if we could just show the spinner all this time until we show the new task list placeholder (if that makes sense?). Or really whatever doesn't cause any huge jumps on the screen.


There are a couple other issue's I noticed, that aren't specific to this PR, but are specific to the experiment:

- `class` used [here](https://github.com/woocommerce/woocommerce-admin/blob/main/client/two-column-tasks/headers/appearance.js#L16) should be `className` I get a warning in my console.
- The extended task list shows up in every task (you can select either one of the paid **Product Types** or install **Google Listings & Ads** to have the extended task list show:
Screen Shot 2022-02-03 at 9 48 26 AM

**Comment by ilyasfoo** [comment]
Thank you for finding out the issues with this, @louwie17! I appreciate it!

>It be nice if we could just show the spinner all this time until we show the new task list placeholder (if that makes sense?). Or really whatever doesn't cause any huge jumps on the screen.

I found out the actual cause is that our lazy components were bubbling up to [Controller's Suspense](https://github.com/woocommerce/woocommerce-admin/blob/98a46ada0a04a73bf763aaa4971bb984d2b4e61c/client/layout/controller.js#L284), thus showing the app-wide Spinner component. I've fixed it in [25faab7](https://github.com/woocommerce/woocommerce-admin/pull/8233/commits/25faab7bd1103e51fb7d145ad357b3fce0319873) by adding a local Suspense to stop it from bubbling up. 

I've also fixed "Your store today" appearing in and out with [4762eb0](https://github.com/woocommerce/woocommerce-admin/pull/8233/commits/4762eb0848e37456073ca3731cdfacc15b83effa).

>class used [here](https://github.com/woocommerce/woocommerce-admin/blob/main/client/two-column-tasks/headers/appearance.js#L16) should be className I get a warning in my console.

Addressed in [1db60db](https://github.com/woocommerce/woocommerce-admin/pull/8233/commits/1db60db2f8f27fb7b998ba8f1724414610dbdf17)

>The extended task list shows up in every task (you can select either one of the paid Product Types or install Google Listings & Ads to have the extended task list show:

Thanks for catching this! Fixed it in [2540f22](https://github.com/woocommerce/woocommerce-admin/pull/8233/commits/2540f2228b17e497f8a5e4f248f30b09ea3fccd6).

Please have another look!

**Review by louwie17 — APPROVED** [review]
Great work @ilyasfoo, thanks so much for the extra updates and fixes, things are looking pretty great now. Nice find on the `Suspense` issue, I wouldn't of guessed that!

Everything seems to test well for me now, so you got my approval. Feel free to merge it, it would be great though if someone else on @woocommerce/ghidorah could still do another round of testing with it on the beta 2 branch. But I am feeling pretty good about this now 💯 💯

**Comment by ilyasfoo** [comment]
Thanks! Yes, we'll do another round of testing with this in beta 2.

**Pull request merged**
